packages/frontend/src/components/search.tsx

```diff
@@ -0,0 +1,43 @@
+import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { memo, useState } from "react";
+
+interface SearchProps {
+  onSubmit: (query: string) => void;
+  showInput: boolean;
+}
+
+export const SearchButton = memo(function SearchButton({
+  onSubmit,
+  showInput,
+}: SearchProps) {
+  const [search, setSearch] = useState("");
+  function onSubmitWrapper() {
+    onSubmit(search);
+  }
+
+  return (
+    <div className="display-flex flex-row">
+      <input
+        hidden={!showInput}
+        onChange={(e) => {
+          setSearch(e.target.value);
+        }}
+        onKeyUp={(e) => {
+          if (e.key === "13") {
```

Comment:
Good job on using `e.key` instead of the other deprecated descriptors.

However, there's a logic error here: I think you meant `"Enter"` here, rather than `"13"` — `"13"` is for `e.which`, which is deprecated.

```suggestion
          if (e.key === "Enter") {
```

Further reading: The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#instance_properties) about key events are an interesting read — you chose `e.key` wisely, but why is that wise? 🤔